Sources/SwiftProtobuf/PathDecoder.swift

```diff
@@ -0,0 +1,426 @@
+    try setValue(&value, defaultValue: .init())
+  }
+
+  mutating func decodeSingularSFixed32Field(value: inout Int32?) throws {
+    try setValue(&value, defaultValue: nil)
+  }
+
+  mutating func decodeRepeatedSFixed32Field(value: inout [Int32]) throws {
+    try setRepeatedValue(&value)
+  }
+
+  mutating func decodeSingularSFixed64Field(value: inout Int64) throws {
+    try setValue(&value, defaultValue: .init())
+  }
+
+  mutating func decodeSingularSFixed64Field(value: inout Int64?) throws {
+    try setValue(&value, defaultValue: nil)
+  }
+
+  mutating func decodeRepeatedSFixed64Field(value: inout [Int64]) throws {
+    try setRepeatedValue(&value)
+  }
+
+  mutating func decodeSingularBoolField(value: inout Bool) throws {
+    try setValue(&value, defaultValue: .init())
+  }
+
+  mutating func decodeSingularBoolField(value: inout Bool?) throws {
+    try setValue(&value, defaultValue: nil)
+  }
+
+  mutating func decodeRepeatedBoolField(value: inout [Bool]) throws {
+    try setRepeatedValue(&value)
+  }
+
+  mutating func decodeSingularStringField(value: inout String) throws {
+    try setValue(&value, defaultValue: .init())
+  }
+
+  mutating func decodeSingularStringField(value: inout String?) throws {
+    try setValue(&value, defaultValue: nil)
+  }
+
+  mutating func decodeRepeatedStringField(value: inout [String]) throws {
+    try setRepeatedValue(&value)
+  }
+
+  mutating func decodeSingularBytesField(value: inout Data) throws {
+    try setValue(&value, defaultValue: .init())
+  }
+
+  mutating func decodeSingularBytesField(value: inout Data?) throws {
+    try setValue(&value, defaultValue: nil)
+  }
+
+  mutating func decodeRepeatedBytesField(value: inout [Data]) throws {
+    try setRepeatedValue(&value)
+  }
+
+  mutating func decodeSingularEnumField<E>(
+    value: inout E
+  ) throws where E : Enum, E.RawValue == Int {
+    try setValue(&value, defaultValue: .init())
+  }
+
+  mutating func decodeSingularEnumField<E>(
+    value: inout E?
+  ) throws where E : Enum, E.RawValue == Int {
+    try setValue(&value, defaultValue: nil)
+  }
+
+  mutating func decodeRepeatedEnumField<E>(
+    value: inout [E]
+  ) throws where E : Enum, E.RawValue == Int {
+    try setRepeatedValue(&value)
+  }
+
+  mutating func decodeSingularMessageField<M>(
+    value: inout M?
+  ) throws where M : Message {
+    if nextPath.isEmpty {
+      try setValue(&value, defaultValue: nil)
+      return
+    }
+    var decoder = try PathDecoder<M>(
+        path: nextPath,
+        value: self.value,
+        mergeOption: mergeOption
+    )
+    if value == nil {
+      value = .init()
+    }
+    try value?.decodeMessage(decoder: &decoder)
+  }
+
+  mutating func decodeRepeatedMessageField<M>(
+    value: inout [M]
+  ) throws where M : Message {
+    try setRepeatedValue(&value)
+  }
+
+  mutating func decodeSingularGroupField<G>(
+    value: inout G?
+  ) throws where G : Message {
+    try setValue(&value, defaultValue: nil)
```

Comment:
Why is this some different than decodeSingularMessageField?  Shouldn't they basically be the same as groups are still messages?